auth/auth.go

```diff
@@ -52,8 +52,20 @@ func ConfigureGithubClient() GithubClient {
 
 	tc := oauth2.NewClient(context.Background(), ts)
 
+	var githubClient *github.Client
+
+	if os.Getenv("GITHUB_HOSTNAME") != "" {
+		GithubHostname := os.Getenv("GITHUB_HOSTNAME")
+		baseUrl := fmt.Sprintf("https://%s/", GithubHostname)
+
+		githubClient, _ = github.NewEnterpriseClient(baseUrl, baseUrl, tc)
```

Comment:
[`NewEnterpriseClient` function](https://github.com/google/go-github/blob/a8da03b06966f3dae0b1476b7d7fe14bacc0ba47/github/github.go#L377) checks out 👌 ... I was initially asking whether this took into account the path aspects for GHES API endpoints, but confirmed that it should be handled appropriately.